packages/frontend/cypress/integration/specs/01-trade-long.spec.js

```diff
@@ -0,0 +1,345 @@
+          cy.get('#open-long-eth-input').clear().type('\\', { delay: 200 }).should('have.value', '0')
+          cy.get('#open-long-osqth-input').clear().type('\\', { delay: 200 }).should('have.value', '0')
+        })
+
+        it('can enter an amount into eth input, before & post trade amount match on position card and osqth input', () => {
+          // cy.get('#open-long-eth-input').clear().type('1', { force: true, delay: 200 }).should('have.value', '1')
+          cy.get('#open-long-eth-input').clear().type('1.', { force: true, delay: 200 }).should('have.value', '1.0')
+          cy.get('#open-long-eth-before-trade-balance').then((bal) => {
+            cy.get('#open-long-eth-post-trade-balance').should('contain.text', (Number(bal.text()) - 1.1).toFixed(2))
+          })
+
+          cy.get('#open-long-osqth-input').should('not.equal', '0')
+          cy.get('#open-long-osqth-input').then((val) => {
+            cy.get('#open-long-osqth-before-trade-balance').then((bal) => {
+              cy.get('#open-long-osqth-post-trade-balance').should(
+                'contain.text',
+                new BigNumber(val.val().toString()).plus(Number(bal.text())).toFixed(2),
+              )
+            })
+            cy.get('#position-card-before-trade-balance').then((bal) => {
+              cy.get('#position-card-post-trade-balance').should(
+                'contain.text',
+                new BigNumber(val.val().toString()).plus(Number(bal.text())).toFixed(2),
+              )
+            })
+          })
+        })
+
+        it('can enter an amount into osqth input, before & post trade amount match on position card and eth input', () => {
+          // cy.get('#open-long-osqth-input').clear().type('1.', { delay: 200 }).should('have.value', '1.0')
+          cy.get('#open-long-osqth-input').clear().type('1.', { force: true, delay: 200 }).should('have.value', '1.0')
+
+          cy.get('#open-long-osqth-before-trade-balance').then((bal) => {
+            cy.get('#open-long-osqth-post-trade-balance').should('contain.text', (Number(bal.text()) + 1.1).toFixed(2))
+          })
+
+          cy.get('#open-long-eth-input').should('not.equal', '0')
+          cy.get('#open-long-eth-input').then((val) => {
+            cy.get('#open-long-eth-before-trade-balance').then((bal) => {
+              cy.get('#open-long-eth-post-trade-balance').should(
+                'contain.text',
+                new BigNumber(bal.text()).minus(val.val().toString()).toFixed(2),
+              )
+            })
+          })
+          cy.get('#open-long-osqth-input').then((val) => {
+            cy.get('#position-card-before-trade-balance').then((bal) => {
+              cy.get('#position-card-post-trade-balance').should(
+                'contain.text',
+                new BigNumber(val.val().toString()).plus(Number(bal.text())).toFixed(2),
+              )
+            })
+          })
+        })
+
+        it('can use max button for open long eth input', () => {
```

Comment:
can we also check here first that the input value is zero before performing the action? so we are sure that it's the action that caused it not to be zero.

Reply:
Oki, will add!